### ISSUE TITLE
Authorize access to tasks

### ISSUE THREAD

**Issue by FrDu** [wekan/wekan#1122]
Authorize access to tasks
User with "normal" and "comment only"  permission should only see the tasks which have been assigned to them by the board admin(s).

**Comment by FrDu** [comment]
Sorry, if used task i mend **card**.

**Comment by gustavooliveir** [comment]
I am very interested in this feature too. To be able to set private/public only to Boards forces you to create many Boards for a large team, which makes a big mess in the Boards screen. Another option would be to group Boards into groups, but that would have to be discussed in another thread.

**Comment by xet7** [comment]
@FrDu @gustavooliveir 

Would it be enough to set board as Hidden, as a new option in the same Board menu Private/Public/Hidden?

So after implementing My Cards #1667 those assigned cards would be visible in each users My Cards, but those users would not be able to see those board(s) at all at All Boards page.

**Comment by gustavooliveir** [comment]
Maybe create tags to show/hide specific boards would be enough.

**Comment by xet7** [comment]
@gustavooliveir 

With tags, do you mean labels?

For organizing All Boards page, currently I'm thinking more about
(nested) tabs and move with multiselect https://github.com/wekan/wekan/issues/2796#issuecomment-590298331

**Comment by Beltrando** [comment]
Hello, I am a newbee here and I do need this feature too. 
I can't contribute, but I can pay some bucks.

**Comment by Beltrando** [comment]
> @FrDu @gustavooliveir
> 
> Would it be enough to set board as Hidden, as a new option in the same Board menu Private/Public/Hidden?
> 
> So after implementing My Cards #1667 those assigned cards would be visible in each users My Cards, but those users would not be able to see those board(s) at all at All Boards page.

@xet7 Hello. I really need this, or some other solution so only assigned cards are visible for an assignee. How can we do that ? I am a newbee, and can't contribute by programming, only by a donation.

**Comment by xet7** [comment]
It is possible to participate funding feature or fix with payment options listed at https://wekan.team/commercial-support/ . All paid features and fixes will be implemented. Current paid upcoming features are listed at Roadmap with some number amount of money, link to Roadmap is at https://wekan.github.io . Highest paid have highest priority.

**Comment by JFa-Orkis** [comment]
On me demande également cette possibilité, avez-vous avancé dessus ?

**Comment by Beltrando** [comment]
> Hello,
> 
> I am also asked for this possibility, have you made progress on it?

xet7 m'a fait une proposition à 1000e pour traiter le besoin, mais pas avant 2 ou 3 mois... On a laissé tomber pour le moment. Et il semblerait bien qu'il ne soit pas possible sur ce projet gitHub de demander à d'autres personnes : il n'y a que lui qui accepte les primes. Mais si vous trouvez qq d'autres je suis preneur.

**Comment by JFa-Orkis** [comment]
xet7 m'a fait une proposition à 1000e pour traiter le besoin, mais pas avant 2 ou 3 mois... On a laissé tomber pour le moment. Et il semblerait bien qu'il ne soit pas possible sur ce projet gitHub de demander à d'autres personnes : il n'y a que lui qui accepte les primes. Mais si vous trouvez qq d'autres je suis preneur.

Ok, **merci pour ce retour**, nous pouvons participer à un financement ponctuel mais à ce niveau mes utilisateurs vont retourner sur Trello en version payante...

Sinon, vous avez des recommandations de fonctionnement pour permettre de cloisonner les accès ?
Les tableaux seraient globalement partagés pour l'équipe, mais nous avons des prestataires externes qui ne devraient accéder qu'aux cartes qui les concerne...

**Comment by Beltrando** [comment]
> Les tableaux seraient globalement partagés pour l'équipe, mais nous avons des prestataires externes qui ne devraient accéder qu'aux cartes qui les concerne...

Je suis exactement dans le même cas que vous .... sauf que ce sont des ingés internes. 

Vous avez vu s'il existe d'autres ingés qui peuvent faire évoluer Wekan ?

**Comment by JFa-Orkis** [comment]
Non, je n'ai pas encore cherché ça

Vous êtes ou ?  ça n'est pas la même mer quoi moi il me semble...  :-D
(nous Marseille)

**Comment by Beltrando** [comment]
> Vous êtes ou ? ça n'est pas la même mer quoi moi il me semble... :-D
> (nous Marseille)
Au siège à Versailles. Et derrière, c'est bien la même mer, mais du côté du plateau technique, à Tunis.
Si vous avez une idée pour Wekan, je suis prêt à participer.

**Comment by mfilser** [comment]
@JFa-Orkis 
@Beltrando 
Please write only in English !

**Comment by xet7** [comment]
According to Google Translate, @JFa-Orkis and @Beltrando seems to be discussing can they afford to pay to me, or do they have developer that could send pull request.

Yes, I am maintainer of Wekan since 2016-12, and only one that can be paid in this GitHub project. Others that did get some payment did disappear after getting payment, they did not continue contributing to Wekan.

I have made new releases of Wekan about every 3rd day since 2016-12, adding many new features and fixes, mostly for free, only some minimal funding, so I would prefer payment to be able to pay my montly bills. Sure there have been also many visiting superstars that have sent features and fixes as pull requests.

**Comment by xet7** [comment]
For me and many others, continuing to use Trello is not an option:
- I did import my boards from Trello to Wekan years ago, I can not afford Trello paid plans with high amount of boards I have
- Trello does not have Swimlanes etc features
- I prefer my data being private at my own computer, not at some propietary cloud service that is under Cloud Act
- For Wekan users at Iran, AWS is blocked, they can not use Trello etc
- Some others can not afford Trello/Jira/etc, Wekan is Open Source, and those that do not have money can improve Wekan with translations and pull requests. Wekan is translated to 60 languages.

**Comment by JFa-Orkis** [comment]
I agree with all of this! 🙂
Afterwards, if paying is possible, it is not the first way to go. We can also be satisfied with the existing one and pray.
Thank you for your investment, and above all, keep going!

(and thank you Google translate too)

**Comment by xet7** [comment]
@JFa-Orkis

Yes, praying works too. I and other contributors are adding new features and fixes very often.

**Comment by Beltrando** [comment]
> Yes, praying works too. I and other contributors are adding new features and fixes very often.
@xet7 According to https://github.com/wekan/wekan/graphs/contributors it seems that quite 99% of commit are yours (2020) ?

**Comment by xet7** [comment]
@Beltrando 

Yes, because I have merged all pull requests, added also code from those anonymous contributors that have not made commits themselves, merged all new translations etc. Wekan did exist by original creator of Wekan, before I continued maintaining Wekan, and very many other contributors have sent big and small features and fixes, so it's not only me coding Wekan.

**Comment by JFa-Orkis** [comment]
Praying is not enough, or your prayers are better than ours... 😉

@Beltrando 
Hello, my management is considering funding this option, would you participate as well? 

@xet7 
Hello Lauri,
Our request :
That some users can only see the cards they are assigned in, the other cards not even being visible.
Addition of a participation as a visitor, without the right to modify, only the possibility of commenting. All over the board or card by card.
This is to manage external service providers, who must be able to access board but only for the cards where they participate. It would be too complex to do cross-board.

Thank you for making us an implementation proposal as soon as possible.

Best regards,
Jacques 

(sorry for my bad english, and thanks google trad )

**Comment by xet7** [comment]
@JFa-Orkis 

Well, I have coded some features and fixes for free (like all security fixes), and received many answers to my prayers.

But funding with real money means me adding this feature to [Roadmap](https://boards.wekan.team/b/D2SzJKZDS4Z48yeQH/wekan-open-source-kanban-board-with-mit-license), me being able to pay some of my real monthly bills (meaning less stress), and more motivation to code this soon.

**Comment by xet7** [comment]
@JFa-Orkis

> Our request :
> That some users can only see the cards they are assigned in, the other cards not even being visible.
> Addition of a participation as a visitor, without the right to modify, only the possibility of commenting. All over the board or card by card.
> This is to manage external service providers, who must be able to access board but only for the cards where they participate. It would be too complex to do cross-board.

Current roles are: BoardAdmin, Normal, CommentOnly, NoComments, Worker

I think this is new role: Read assigned only Comment only.

It would would work like other roles, in web UI right sidebar click avatar change role, and in REST API.

It would be released included in new version of Wekan.

If it is only this feature, it is 500 euro. After pre-payment is received, possible delivery date is 2 months from that received date, or earlier.

Alternative is if I sometime get funding for Granular Roles #3022 that is bigger feature about 2000 euro. After pre-payment is received, possible delivery date is 4 months from that received date, or earlier.

Payment options are at https://wekan.team/commercial-support/ . Usually companies request invoice PDF. Orders and more details at support@wekan.team .

With "possible delivery dates" I do mean it's possible, as I can only guess, but I can not see into future. It is also possible that something causes it to be later. Or it can also be much earlier.

**Comment by xet7** [comment]
@JFa-Orkis 

Yes, in that case of NormalAssignedOnly, only difference is that only Assigned cards are visible.

**Comment by JFa-Orkis** [comment]
Perfect, I check internally that we are in phase, and I come back to you

**Comment by JFa-Orkis** [comment]
Ok, we validate this request, I'm looking to make the prepayment with option 3

Have a good day,

**Comment by JFa-Orkis** [comment]
Transfer of 500 € to EEZY made today!

**Comment by xet7** [comment]
@JFa-Orkis 

Thanks! I added this [to Roadmap](https://boards.wekan.team/b/D2SzJKZDS4Z48yeQH/wekan-open-source-kanban-board-with-mit-license/so45bWgr8xntnEvDJ).

**Comment by JFa-Orkis** [comment]
Yes, we understood that correctly, but 6 months have already passed ...
There will always be other urgent features that are going to be added, and I don't think I'm going overboard and asking the question after 6 months of waiting.
Good day, hoping for a breakthrough before the end of 2022
Jacques

**Comment by kvacha13** [comment]
Very important feature! Any update on this?

**Comment by Beltrando** [comment]
Hello @JFa-Orkis Vous avez reçu des nouvelles ?

**Comment by xet7** [comment]
@Beltrando @JFa-Orkis 

I need to do speedups to WeKan first, so that I would not need to implement this assigned feature twice, there is not enough funding to implement twice. Speedups require major internal changes.

After speedups I try to do this assigned feature, during 2022.

**Comment by Beltrando** [comment]
I don't understand the connection between https://github.com/wekan/wekan/issues/4061 and https://github.com/wekan/wekan/issues/1122, but I do understand that you told @JFa-Orkis on March 9, 2021 that it would be done in 2 months if there was a 500e payment. And that on March 15 he paid that 500e.

We are not 2 months later but 15 months later. And if I understand correctly it is still not done?

**Comment by xet7** [comment]
@Beltrando 

Yes, this is still not done. There are many reasons how developing some WeKan feature can get delayed:
- Urgent security fixes to do
- Someone hurrying up development of some feature, causing more stress, and delaying developing everything
- As total, there being not enough WeKan funding to pay my monthly bills, causing me to require to do other non-WeKan work to pay those bills, delaying developing WeKan
- Someone close relative requiring immediate care

What would speed up WeKan development:
- Having more funding at https://wekan.team/commercial-support/ , so that I can pay someone else to do non-WeKan related stuff, and spend more time developing WeKan, or even hire more help to do some WeKan related tasks
- Have more pull requests to WeKan for fixes, etc, having community help, using each of community members special skills. For example, those that know about LDAP, could add fixes to LDAP. I don't have LDAP server.
- Having more encouragement to develop WeKan

In general, I do develop every paid feature, if given enough resources and encouragement to do so.

**Comment by xet7** [comment]
@JFa-Orkis 

I'm developing new version of WeKan, because not all new features are possible with current Meteor WeKan. All paid features will be implemented in that new version.

**Comment by kvacha13** [comment]
@xet7 Lauri Ojansivu 
Any updates on this issue? On roadmap it says 01/03 ? Can you please update the date on roadmap?

**Comment by Beltrando** [comment]
@xet7 

Hello,

In a message in February 2021 above, you had written: "Yes, I am maintainer of Wekan since 2016-12, and only one that can be paid in this GitHub project. Others that did get some payment did disappear after getting payment, they did not continue contributing to Wekan."

Of course you have not disappeared, I hope, but you have received payment for this work for 2 years and nothing has been delivered. 

An answer would be appreciated.

**Comment by Beltrando** [comment]
@xet7 
You told @JFa-Orkis on March 9, 2021 that it would be done in 2 months if there was a 500e payment. On March 15 he paid that 500e.

We are not 2 months later but almost 3 years later.

**Comment by xet7** [comment]
@Beltrando 

Yes, that's correct. I'm adding all paid features. Yes, it takes some time.

**Comment by Beltrando** [comment]
Hello, so this may take a little time. Okay.

But it's been since *March 2021*. 3 years and a half after the payment,
it's still not done !

What do you think?

Kind regards,

Eric Beltrando
ELAU - 11 rue de Lourmel, Paris
www.linkedin.com/in/Éric <http://www.linkedin.com/in/%C3%A9ric>
***@***.***

Nos *528 projets*: http://www.bilog.fr/realisations
Standard:  +331 30 21 63 92



Le mer. 17 janv. 2024 à 10:55, Lauri Ojansivu ***@***.***> a
écrit :

**Comment by xet7** [comment]
@Beltrando 

Yes, I can not stop time going forwards.

I can only add new features, one by one, until all are added.

**Comment by Beltrando** [comment]
> @Beltrando
> 
> Yes, I can not stop time going forwards.
> 
> I can only add new features, one by one, until all are added.



A long lng time ago you wrote : 
[xet7](https://github.com/xet7) commented [on Feb 19, 2021](https://github.com/wekan/wekan/issues/1122#issuecomment-781823884)
... Yes, I am maintainer of Wekan since 2016-12, and only one that can be paid in this GitHub project. **Others that did get some payment did disappear after getting payment**, they did not continue contributing to Wekan.

**Well, 3 and a half years after receiving this money, you still haven't done anything.
What name do you give to someone you've paid for a service, and who 3 years later hasn't done it?**

**Comment by xet7** [comment]
Please, everyone, try to be encouraging and friendly to all FOSS maintainers, so that FOSS maintainers would continue maintaining and adding features and fixes. For example, for LDAP login, WeKan uses old version of LDAP npm package, that is not maintained anymore, because someone was not friendly to maintainer of that LDAP package: https://github.com/ldapjs/node-ldapjs/blob/8ffd0bc9c149088a10ec4c1ec6a18450f76ad05d/README.md

Currently, for WeKan, I'm figuring out how to update WeKan dependencies, and add all paid features.

**Comment by Beltrando** [comment]
How can we provide funding when this was done 3 years ago, and you still haven't dealt with the issue?

By the way, I've just emailed you my proposal for 2020.

**Comment by xet7** [comment]
@Beltrando 

Please stop repeating yourself. Really, I will add this feature during 2025.

Today is Christmas Eve.

**Comment by Beltrando** [comment]
@xet7 As requested I "stop repeating yoursel".

Eleven months have passed.

Still nothing.

Sigh

**Comment by xet7** [comment]
@Beltrando 

Yes, negative comments slow me down. Thanks for your comment!
